lib/checkstl.cpp

```diff
@@ -3152,14 +3152,13 @@ void CheckStl::eraseIteratorOutOfBounds()
             if (!tok->valueType())
                 continue;
             const Library::Container* container = tok->valueType()->container;
-            if (!container)
-                continue;
-            if (!Token::Match(tok->next(), ". %name% ("))
+            if (!container || !Token::simpleMatch(tok->astParent(), "."))
                 continue;
-            const Library::Container::Action action = container->getAction(tok->strAt(2));
+            const Token* const ftok = tok->astParent()->astOperand2();
+            const Library::Container::Action action = container->getAction(ftok->str());
```

Comment:
probably does not really matter but it seems to me this will frequently be used when ftok is tok because you didn't check if tok != ftok

     y = data.x;  // <- if x is a container then we will lookup what action method "data.x.x(..)" would have